src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -679,8 +689,11 @@ public override void VisitParameter(IParameterSymbol symbol)
             // used on their own or in the context of methods.
 
             var includeType = format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeType);
+
             var includeName = format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeName)
-                              && !(symbol.ContainingSymbol is IMethodSymbol { MethodKind: MethodKind.FunctionPointerSignature });
+                && (symbol is Symbols.PublicModel.Symbol { UnderlyingSymbol: SignatureOnlyParameterSymbol } // SignatureOnlyParameterSymbol.ContainingSymbol throws NotSupportedException
```

Comment:
>&& (symbol is Symbols.PublicModel.Symbol { UnderlyingSymbol: SignatureOnlyParameterSymbol } // SignatureOnlyParameterSymbol.ContainingSymbol throws NotSupportedException [](start = 16, length = 169)

I guess the motivation for this change is not clear to me. I was suggesting to revert to the original behavior:
```
            var includeName = format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeName)
                              && !(symbol.ContainingSymbol is IMethodSymbol { MethodKind: MethodKind.FunctionPointerSignature });
``` 
unless there is a specific reason.

Reply:
The reason is that the original code can throw, as I discovered when I moved that expression to a local variable that ran unconditionally. We just happen not to have any tests where we use `IncludeName` with a `SignatureOnlyParameterSymbol`

Reply:
> The reason is that the original code can throw, as I discovered when I moved that expression to a local variable that ran unconditionally. We just happen not to have any tests where we use IncludeName with a SignatureOnlyParameterSymbol

We need to back this change with a targeted unit-test then. The test should break with the original code here. Also, we should consider adjusting the check by looking at the root cause of the problem that the ```!(symbol.ContainingSymbol is IMethodSymbol { MethodKind: MethodKind.FunctionPointerSignature })``` condition was added to address. I mean the fact that display was "bad" when the parameter had an empty name. I think we should simply check for an empty name then, rather than trying to detect that indirectly. I am thinking of something like that:
``` 
var includeName = format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeName)
                  && !symbol.Name.IsNullOrEmpty();
```

---
In reply to: [592339443](https://github.com/dotnet/roslyn/pull/51223#discussion_r592339443) [](ancestors = 592339443)

Reply:
That ends up triggering a huge number of changes, since in error cases things often don't have names.

It breaks a lot of tests, but also means that we don't print the default value of an optional parameter when the parameter doesn't have a name.

Reply:
> That ends up triggering a huge number of changes, since in error cases things often don't have names.

> It breaks a lot of tests, but also means that we don't print the default value of an optional parameter when the parameter doesn't have a name.

Could you give some examples of these behavior changes? Used to be this, now this. Also some examples of what are those parameters without names, where are they coming from.

---
In reply to: [592362609](https://github.com/dotnet/roslyn/pull/51223#discussion_r592362609) [](ancestors = 592362609)